server/index.js

```diff
@@ -35,6 +35,31 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId, 10);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return res.sstatus(400).json({
+      error: '"productId" must be a positive integer'
+    });
+  }
+  const sql = `
+  select *
+    from "products"
+    where "productId" = $1
+    `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`Cannot find product with "productId" ${productId}`, 404));
+      } else {
+        return res.status(200).json(result.rows);
```

Comment:
should return `product` here not `result.rows` that will put your product in an array which will suck for your front end.

Reply:
Fixed!